code/script.js

```diff
@@ -222,104 +226,169 @@ const generateBoard = () => {
+    )
+    charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
+    } else if (category === 'eyes') {
+    if (keep) {
+     alert(
+     `Yes, the person has ${value} eyes! Keep all people with ${value} eyes.`
+    )
+    charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else { 
+     alert(
+     `No, the person doesn't have ${value} eyes! Remove all people with ${value} eyes.`
+    )
+    charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+// This function finally redraws the board with the remaining people
+generateBoard()   
+}
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+// when clicking guess, the eventlistener makes the confirm-popup appear to confirm that you want to make a guess, 
+//and stores the guess from the player in the personToConfirm variable. Then CheckMyGuess is invoked.
+
+const guess = (personToConfirm) => {
+  const confirmOk = confirm(
+    `Are you sure about this guess of ${personToConfirm}?`
+  )
 
-  // Invoke a function to redraw the board with the remaining people.
+if (confirmOk) {  
+checkMyGuess(personToConfirm)
+}
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+// when you confirm, this function is invoked and depending on if your guess is right or wrong you are directed to win or lose page
+const checkMyGuess = (personToConfirm) => {
+
+if (personToConfirm === secret.name) {
+  winOrLoseText.innerHTML+= "You won the game! 🕵️✨"
+} else {
+  winOrLoseText.innerHTML+= "You lost the game! ❌🙅"
 }
 
-// If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+//here we show the WinOrLose section and hide the game board
+winOrLose.style.display = "block"
+board.style.display = "none"
 }
 
+// this section is for clicking the restart button, not working unfortunately
+playAgainButton.addEventListener = ('click', (start) => {
```

Comment:
It seems you have tried to make a function out of an addEventlistener. 
Try playAgainButton.addEventListener('click', start) where "start" is the function that is invoked when you click on the playAgainButton. Try moving your code (row 379 - 380) to your start function instead. Hopefulle this will solve the problem.